### DISCUSSION TITLE
programming button of a dual R3

### DISCUSSION THREAD

**Discussion by Ansgar66** [arendst/Tasmota#12361] [Ideas]
Hi all,
I just managed to control my shutter with help of a dual R3 and calibrated the whole thing.
in addition to control via web interface I have two buttons connected to the button entries s1 and s2.
usage of these two buttons require to hold it the whole time I want to drive the shutters.
I read there's an opportunity to change and make a doubleclick drive the curtain completely down or up, but I don't find the right command to send at the console. 
can anybody help me here?

thanks
Ansgar

**Comment by barbudor** [comment]
Have you looked at that ? https://tasmota.github.io/docs/Blinds-and-Shutters/#button-control and that https://tasmota.github.io/docs/Commands/#shutterbutton

The `ShutterButton` should help you

**Comment by Ansgar66** [comment]
Dear Barbudor,

thanks a lot for your quick response.

And yes. I tried ShutterButton<x> <button> up <mqtt>, resp. ShutterButton<x> <button> down <mqtt> but it’s unclear to me how to use it.

In case of the dual R3, I only have one shutter connected. So <x> is irrelevant or 1, I understood.

But what do I have to enter for <button> and <mqtt>?

Following your tip   <https://tasmota.github.io/docs/Commands/#shutterbutton> https://tasmota.github.io/docs/Commands/#shutterbutton I tried Backlog ShutterButton1 1 up 1; ShutterButton1 2 down 1,

The accompanying text exactly describes my task, but the command lead to my shutter not reacting to the buttons at all !?

 

Any idea?

 

Thanks

Ansgar

 

 

 

Von: Barbudor ***@***.***> 
Gesendet: Sonntag, 13. Juni 2021 19:41
An: arendst/Tasmota ***@***.***>
Cc: Ansgar66 ***@***.***>; Author ***@***.***>
Betreff: Re: [arendst/Tasmota] programming button of a dual R3 (#12361)

 

Have you looked at that ? https://tasmota.github.io/docs/Blinds-and-Shutters/#button-control and that https://tasmota.github.io/docs/Commands/#shutterbutton

The ShutterButton should help you

**Reply by barbudor** [reply]
You may have not read the 1st link totally : https://tasmota.github.io/docs/Blinds-and-Shutters/#button-control
`ShuuterButton` has 2 syntax. The one you are using with 3 arguments, and the 2nd one 👍 
```
ShutterButton<x> <button> <p1> <p2> <p3> <ph> <m1> <m2> <m3> <mh> <mi>

<button> 1..4: Button number, 0/-: disable buttons for this shutter
<p1> 0..100: single press position, t: toggle, -: disable
<p2> 0..100: double press position, t: toggle, -: disable
<p3> 0..100: tripple press position, t: toggle, -: disable
<ph> 0..100: hold press position, t: toggle, -: disable
<m1> 1: enable single press position MQTT broadcast, 0/-: disable
<m2> 1: enable double press position MQTT broadcast, 0/-: disable
<m3> 1: enable tripple press position MQTT broadcast, 0/-: disable
<mh> 1: enable hold press position MQTT broadcast, 0/-: disable
<mi> 1: enable MQTT broadcast to all shutter indices, 0/-: disable

Parameters are optional. When missing, all subsequent parameters are set to disable.
```

You can use that to assign shutter positions to single-press, double-press, triple-press and hold.
If you don't need specific MQTT message then ignore the remaining parameters

**Comment by Ansgar66** [comment]
Hi Barbudor, thanks again. 

(the text below is second try; I didn't know I must use some escape letter to use "<" or ">", so I replaced them with ')

I tried the following to solve my problem:

1.) Command ShutterButton'x' 'button' up/down 'mqtt'
I may dismiss 'x' since the dual R3 can only address one single shutter
'button'= 1; I translated with the number of the physical terminal S1
'mqtt'= 1; since I want to get mqtt messages related to shutteractions

the result on sending the command
ShutterButton 1 up 1
was
{"ShutterButton1":"1 100 50 74 100 - - - 1 1"}

the result on sending the command
ShutterButton 2 down 1
was
{"ShutterButton1":"1 100 50 74 100 - - - 1 1|2 0 50 24 0 - - - 1 1"}

looked perfect, but the reaction on using the buttons remains the same: The shutter only moves as long as I push it.

2.) Second try using the command
ShutterButton 1 99 1 75 99 - - - 1 1
results in
{"ShutterButton1":"1 98 0 74 98 - - - 1 1|2 0 50 24 0 - - - 1 1"}

Also looks reasanoble despite numbers were only nearly but not exactly taken over, but still no change in "pushbuttonbehaviour"!?!

what do I do wrong???

**Comment by Ansgar66** [comment]
Hi Barbudor,
it's me again; unfortunately, I got stucked at the point described above:
the software aknowleges my commands reasanobly, but the appropriate change of behaviour doesn't happen.
Would you recommend a software update to the next version to give it another try`?

Thanks
Ansgar

**Reply by barbudor** [reply]
Sorry I'm not well in Shutter, haven't used it myelf.
I was mainly pointing you to the right documentation to assign different open/close % to different button actions
